src/main/webapp/app/entities/quiz-exercise/quiz-exercise-detail.component.ts

```diff
@@ -555,53 +605,77 @@ export class QuizExerciseDetailComponent implements OnInit, OnChanges, OnDestroy
         return isGenerallyValid && areAllQuestionsValid;
     }
 
+    /**
+     * @function computeInvalidWarnings
+     * @desc Get the reasons, why the quiz needs warnings
+     * @returns {Array} array of objects with fields 'translateKey' and 'translateValues'
+     */
+    computeInvalidWarnings(): Warnings[] {
+        let invalidWarnings = new Array<Warnings>();
```

Comment:
This section is more typical to java than to js/ts.
Instead of creating a let and then pushing into it, it would be better to create a const and directly construct the array.

This would look like this:

const invalidWarnings = !this.quizExercise ? [] : this.quizExercise.quizQuestions.map((question, index) => {
// return result of function call.
});

Reply:
Updated.